modules/performing-a-backup-oadp-rosa-sts.adoc

```diff
@@ -0,0 +1,153 @@
+
+Either DPA configuration will work.
+
+. Create a workload to back up by running the following commands:
++
+[source,bash]
+----
+$ oc create namespace hello-world
+----
++
+[source,bash]
+----
+$ oc new-app -n hello-world --image=docker.io/openshift/hello-openshift
+----
+
+. Expose the route by running the following command:
++
+[source,bash]
+----
+$ oc expose service/hello-openshift -n hello-world
+----
+
+. Check the application is working by running the following command:
++
+[source,bash]
+----
+$ curl `oc get route/hello-openshift -n hello-world -o jsonpath='{.spec.host}'`
+----
++
+[source,bash]
+----
+Hello OpenShift!
+----
+
+
+. Back up the workload by running the following command:
++
+[source,bash]
+----
+$ cat << EOF | oc create -f -
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: hello-world
+  namespace: openshift-adp
+spec:
+  includedNamespaces:
+  - hello-world
+  storageLocation: ${CLUSTER_NAME}-dpa-1
+  ttl: 720h0m0s
+EOF
+----
+
+. Wait until the backup is completed and then run the following commands:
++
+[source,bash]
+----
+$ watch "oc -n openshift-adp get backup hello-world -o json | jq .status"
+----
++
+[source,json]
+----
+{
+  "completionTimestamp": "2022-09-07T22:20:44Z",
+  "expiration": "2022-10-07T22:20:22Z",
+  "formatVersion": "1.1.0",
+  "phase": "Completed",
+  "progress": {
+    "itemsBackedUp": 58,
+    "totalItems": 58
+  },
+  "startTimestamp": "2022-09-07T22:20:22Z",
+  "version": 1
+}
+----
+
+. Delete the demo workload by running the following command:
++
+[source,bash]
+----
+$ oc delete ns hello-world
+----
+
+. Restore the workload from the backup by running the following command:
++
+[source,bash]
+----
+$ cat << EOF | oc create -f -
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: hello-world
+  namespace: openshift-adp
+spec:
+  backupName: hello-world
+EOF
+----
+
+. Wait for the Restore to finish by running the following commands:
++
+[source,bash]
+----
+$ watch "oc -n openshift-adp get restore hello-world -o json | jq .status"
+----
++
+....
+{
+  "completionTimestamp": "2022-09-07T22:25:47Z",
+  "phase": "Completed",
+  "progress": {
+    "itemsRestored": 38,
+    "totalItems": 38
+  },
+  "startTimestamp": "2022-09-07T22:25:28Z",
+  "warnings": 9
+}
+....
+
+. Check the workload is restored by running the following command:
++
+[source,bash]
+----
+$ oc -n hello-world get pods
+----
+
++
+[source,bash]
+----
+NAME                              READY   STATUS    RESTARTS   AGE
+hello-openshift-9f885f7c6-kdjpj   1/1     Running   0          90s
+----
++
+[source,bash]
+----
+$ curl `oc get route/hello-openshift -n hello-world -o jsonpath='{.spec.host}'`
+----
++
+[source,bash]
+----
+Hello OpenShift!
+----
+
+. For troubleshooting tips, see the OADP team’s link:https://github.com/openshift/oadp-operator/blob/master/docs/TROUBLESHOOTING.md[troubleshooting documentation].
```

Comment:
I can't link in the documentation to Git.
Are there published documents that I can use instead or should I remove these links?

Reply:
https://red.ht/oadp-faq

Does this work?

Reply:
Yes, thank you!